kie-internal/src/main/java/org/kie/internal/pmml/PMMLImplementationsUtil.java

```diff
@@ -0,0 +1,93 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.internal.pmml;
+
+import org.kie.api.pmml.PMMLConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.kie.api.pmml.PMMLConstants.KIE_PMML_IMPLEMENTATION;
+import static org.kie.api.pmml.PMMLConstants.LEGACY;
+import static org.kie.api.pmml.PMMLConstants.NEW;
+
+/**
+ * Class used to provide utility methods to manage implementation to be invoked
+ * at runtime
+ */
+public class PMMLImplementationsUtil {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PMMLImplementationsUtil.class);
+    private static final String LEGACY_IMPL = "org.kie.pmml.assembler.PMMLAssemblerService";
+    private static final String TRUSTY_IMPL = "org.kie.pmml.evaluator.assembler.service.PMMLAssemblerService";
+
+    /**
+     * @param classLoader
+     * @return the <b>PMML</b> implementation to enable
+     */
+    public static PMMLConstants toEnable(final ClassLoader classLoader) {
+        boolean isLegacyPresent = isImplementationPresent(LEGACY_IMPL, classLoader);
+        boolean isTrustyPresent = isImplementationPresent(TRUSTY_IMPL, classLoader);
+        if (!isLegacyPresent && !isTrustyPresent) {
+            throw new IllegalStateException("Could not find PMML implementation");
+        }
+        PMMLConstants pmmlConstants = PMMLConstants.byName(System.getProperty(KIE_PMML_IMPLEMENTATION.getName(), LEGACY.getName()));
+        switch (pmmlConstants) {
+            case LEGACY:
+                if (isLegacyPresent) {
+                    return LEGACY;
+                } else {
+                    return NEW;
+                }
+            case NEW:
+                if (isTrustyPresent) {
+                    return NEW;
+                } else {
+                    return LEGACY;
+                }
+            default:
+                throw new RuntimeException("Unmanaged PMMLConstants " + pmmlConstants);
+        }
+    }
+
+    /**
+     * @param classLoader
+     * @return <code>true</code> if the <b>implementationFullName</b> is found in the given <code>ClassLoader</code>,
+     * <code>false</code> otherwise
+     */
+    private static boolean isImplementationPresent(final String implementationFullName, final ClassLoader classLoader) {
+        try {
+            classLoader.loadClass(implementationFullName);
+            return true;
+        } catch (NoClassDefFoundError | ClassNotFoundException e) {
+            return false;
+        }
+    }
+
+    /**
+     * @param classLoader
+     * @return <code>true</code> if <b>org.kie.dmn.jpmml.DMNjPMMLInvocationEvaluator</b> is found in the given <code>ClassLoader</code>,
+     * <code>false</code> otherwise
+     */
+    public static boolean isjPMMLAvailableToClassLoader(final ClassLoader classLoader) {
```

Comment:
Is this called somewhere? Can you point me to the switch for jpmml implementation?

Reply:
@jiripetrlik 
It is used in both PMMLAssemblerService (legacy and new) in my drools DROOLS-5867 branch. And it is meant to be used anywher such info is needed

Reply:
@gitgabrio 
Thank you for explanation.